lib/net.js

```diff
@@ -834,6 +834,15 @@ function connect(self, address, port, addressType, localAddress, localPort) {
 }
 
 
+// Check that the port number is not NaN when coerced to a number,
+// is an integer and that it falls within the legal range of port numbers.
+function isLegalPort(port) {
+  if (typeof port === 'string' && port.trim() === '')
+    return false;
```

Comment:
Empty string and strings of 1 or more spaces become 0 when the + operator is applied. Since we want to allow zero, I had to specifically check for that case.